### PR TITLE
Fixing a bug when "async_switch" option is used.

### PR DESCRIPTION
**Pull request by s-code**
When "async_switch" option is used "$routerArguments" should be passed to the decorated router, but not to the decorator.

### CODE REVIEW

**Comment by UFOMelkor** [comment]
I did not use the async switch, but it looks good to me :+1: 

Do you think you can add a test case for it?
Travis notes about a coding style. Please take a look at the order of the use statements.

The other error is not related to this PR. Feel free to look at it if you want, otherwise I will do.

Also: Is there anything that prevents the usage of `has` to check whether the async switch is enabled instead of using try-catch?

**Comment by s-code** [comment]
@UFOMelkor i've added test cases and replaced try-catch by `has` checking.

**Comment by UFOMelkor** [comment]
The failures in CI have other reasons, will be fixed with #67 .
Unfortunately your test does not cover the issue you mentioned (remove your fix and the tests will still pass).


The bug just appears if the handlers are registered with tags.

I had a look at it and modified as following.
Does this reflect the setup in your application?

```yaml
prooph_service_bus:
  command_buses:
    command_bus_async:
      router:
        async_switch: async_message_producer

services:
  Acme\RegisterUserHandler:
     class: ProophTest\Bundle\ServiceBus\DependencyInjection\Fixture\Model\AcmeRegisterUserHandler
     public: true
     tags: [{name: 'prooph_service_bus.command_bus_async.route_target', message_detection: true }]
  async_message_producer:
     class: ProophTest\Bundle\ServiceBus\DependencyInjection\Fixture\Model\AsyncMessageProducer
     public: true
```

```php
    /**
     * @test
     */
    public function it_creates_a_command_bus_with_async_switch_message_router()
    {
        $container = $this->loadContainer('command_bus_async', new PluginsPass(), new RoutePass());

        /** @var AsyncMessageProducer $asyncMessageProducer */
        $asyncMessageProducer = $container->get('async_message_producer');

        /** @var CommandBus $commandBus */
        $commandBus = $container->get('prooph_service_bus.command_bus_async');

        /** @var AcmeRegisterUserHandler $handler */
        $handler = $container->get('Acme\RegisterUserHandler');

        // dispatch an async command
        $commandBus->dispatch(new AsyncAcmeRegisterUserCommand([]));
        self::assertInstanceOf(AsyncAcmeRegisterUserCommand::class, $asyncMessageProducer->lastMessage);

        // dispatch an sync command
        $commandBus->dispatch(new AcmeRegisterUserCommand([]));
        self::assertInstanceOf(AcmeRegisterUserCommand::class, $handler->lastCommand());
    }
```

**Comment by s-code** [comment]
@UFOMelkor I've made a mistake in fixtures for event bus. Now it covers the issue.
Your modification reflects the setup of our app. Should i apply this changes in my pull request and remove my event bus fixtures?

**Comment by s-code** [comment]
@UFOMelkor i've decided to remove event_bus_async fixtures, so now only command_bus_async fixture is changed

**Comment by UFOMelkor** [comment]
> Your modification reflects the setup of our app. Should i apply this changes in my pull request and remove my event bus fixtures?

Would be great :+1:

**Comment by s-code** [comment]
@UFOMelkor Done, could you check?

File: src/DependencyInjection/Compiler/RoutePass.php [hunk 1]
```diff
@@ -42,7 +42,12 @@ public function process(ContainerBuilder $container)
             $buses = $container->getParameter('prooph_service_bus.' . $type . '_buses');
 
             foreach ($buses as $name => $bus) {
-                $router = $container->findDefinition(sprintf('prooph_service_bus.%s.router', $name));
+                $routerServiceId = sprintf('prooph_service_bus.%s.decorated_router', $name);
+                if (!$container->has($routerServiceId)) {
```
Comment (UFOMelkor) [inline src/DependencyInjection/Compiler/RoutePass.php]: Missing space after `!`

**Review by UFOMelkor — APPROVED** [review]
Just one space missing, then we are ready :-)

**Comment by s-code** [comment]
done, thx

**Comment by UFOMelkor** [comment]
:+1:

**Comment by UFOMelkor** [comment]
Thank you! :tada:

**Pull request merged**